kore/src/main/scala/org/kframework/parser/kore/parser/TextToKore.scala

```diff
@@ -73,23 +85,32 @@ class TextToKore(b: Builders = DefaultBuilders) {
       parseDefinition()
     } catch {
       case e: java.io.EOFException => throw new ParseError("ERROR: Unexpected end of file while parsing", e)
-      case exc: ParseError => throw exc
-      case exc: Throwable => throw new ParseError("ERROR: Unexpected error while parsing: " + exc.getMessage, exc) // shouldn't be reachable
     } finally {
       scanner.close()
     }
   }
 
-  /** Parses from the stream and returns [[kore.Definition]]. */
+  /** Parses from the stream and returns [[kore.Pattern]]. */
   @throws(classOf[ParseError])
   def parsePattern(src: io.Source, line: Integer): Pattern = {
     try {
       scanner.init(src, line)
       parsePattern()
     } catch {
       case e: java.io.EOFException => throw new ParseError("ERROR: Unexpected end of file while parsing", e)
-      case exc: ParseError => throw exc
-      case exc: Throwable => throw new ParseError("ERROR: Unexpected error while parsing: " + exc.getMessage, exc) // shouldn't be reachable
```

Comment:
Why were these cases deleted?

Reply:
the first one isn't needed if not for the second one, and the second one is a bit too broad. It's generally a bad idea to catch anything that extends java.lang.Error, for example, because these exceptions typically represent serious failures of the JVM and are often not recoverable at all.

Reply:
Got it! Thanks!